### DISCUSSION TITLE
Problem calibrating sensor

### DISCUSSION THREAD

**Discussion by giuseppebaldi** [tronikos/esphome-magnetometer-water-gas-meter#46] [General]
I've got this put together, installed and working (temp reported, I can see half rotations and magnetic field strengths). The issue I am having is that calibrating the sensor fails each time with the logs showing this:

```
[18:43:30][I][calibration:188]: Starting calibration. Make sure gas is running
[18:43:41][I][calibration:205]: x ranged from 11.16 to 25.20
[18:43:43][I][calibration:211]: y ranged from -68.43 to -52.15
[18:43:43][I][calibration:217]: z ranged from -41.38 to -29.45
[18:43:43][E][calibration:235]: Calibration failed. No axis had a range of at least 20
```
I have my fireplace running open nearly all the way.  I've tried longer calibration times and also running some other gas appliances along with the fireplace.

I have my volume per half rotation set to 0.00125 CCF as suggested but I'm not 100% sure that's right. I have a Sensus R-275 gas meter, specs [here](https://www.xylem.com/en-us/products--services/metrology-equipment-for-utilities/meters/r-275-diaphragm-meters/). I could not find any information about volume per half-rotation for this meter but according to this page, and others I've found for this meter, the Revolutions is 8 cubic feet. I assume a revolution is the same as a rotation so would the volume per half rotation be 4 or 0.04 CCF? This seems way off of the 0.00125 suggested in the documentation.

**Comment by tronikos** [comment]
Try repositioning the sensor to see if you can get a bigger range. Keep in mind for gas meters the minimum range of 20 isn't often enough and you have to follow the manual calibration part of the README.

Same for your last question. If 0.00125 CCF isn't the right value, follow the README to find the right value.

**Comment by giuseppebaldi** [comment]
Thank you, actually moving the sensor down a few inches on the meter made all the difference in the world, it calibrated on the first try and seems to be reading accurately now. I did go ahead with the suggested volume/half-rotation of 0.00125, that was correct and should be for most, if not all, US gas meters. Thanks for your awesome work on this project!

**Reply by tronikos** [reply]
Would you mind taking a picture and post it here so that I can include it in the readme?

**Comment by giuseppebaldi** [comment]
Sorry, been a bit busy. See attached photos. I originally had it higher up, right under the meter gauges but moving it down onto the indented part of the meter worked MUCH better.

![PXL_20241225_144057647](https://github.com/user-attachments/assets/52268b41-8156-4b82-93c0-e508b34749af)
![PXL_20241225_144042078](https://github.com/user-attachments/assets/be14e906-8301-4273-80d1-900f0a88a23e)
![PXL_20241225_144050054](https://github.com/user-attachments/assets/aa932003-3b47-442f-887b-cc880e595acc)
